### ISSUE TITLE
Non-recoverable error - zonelist.lua attempt to compare nil with number

### ISSUE THREAD

**Issue by NickMcGettigan** [EarendelDevelopers/factorio-mods#255]
Just finished several sciences to unlock more zones / planets.  Attempted to bring up universe explorer but it crashed instead.

Screenshot of error:
![image](https://user-images.githubusercontent.com/20891567/203876527-95e80a6c-3fee-4431-9224-5d984c8367bc.png)

Link to save: https://drive.google.com/file/d/1mRsIJ_lnbf0Mxnw_RryopjEReN3psaaN/view?usp=share_link

Console:
[2022-11-24 15:58:58]: __space-exploration__/scripts/zonelist.lua:1476: in function <__space-exploration__/scripts/zonelist.lua:1474>[2022-11-24 15:58:58] [C]: : in function 'sort'[2022-11-24 15:58:58]: __space-exploration__/scripts/zonelist.lua:1485: in function 'update_list'[2022-11-24 15:58:58]: __space-exploration__/scripts/zonelist.lua:2273: in function 'update'[2022-11-24 15:58:58]: __space-exploration__/scripts/zonelist.lua:1210: in function 'open'[2022-11-24 15:58:58]: __space-exploration__/scripts/zonelist.lua:884: in function 'toggle'[2022-11-24 15:58:58]: __space-exploration__/scripts/zonelist.lua:2732: in function 'callback'[2022-11-24 15:58:58]: __space-exploration__/scripts/event.lua:15: in function <__space-exploration__/scripts/event.lua:13>"[2022-11-24 15:58:58] [46727.538]: Info ServerMultiplayerManager.cpp:795: updateTick(197996472) changing state from(InGame) to(Failed)[2022-11-24 15:58:58] [46727.551]: Quitting: multiplayer error.[2022-11-24 15:59:01] [46729.847]: Info ServerMultiplayerManager.cpp:141: Quitting multiplayer connection.

**Comment by NickMcGettigan** [comment]
Link to log: https://drive.google.com/file/d/1uqIC85pr69V46r-lIfB4DtQ9idGR_mGo/view?usp=sharing

**Comment by NickMcGettigan** [comment]
Link to log: https://drive.google.com/file/d/1uqIC85pr69V46r-lIfB4DtQ9idGR_mGo/view?usp=sharing

**Comment by InappropriatePenguin** [comment]
@NickMcGettigan, did you build a spaceship console on a non-SE surface (such as inside the Editor extensions lab or on a Factorissimo surface)? If so, mine the console from that surface and you should no longer encounter this crash.

The next version of SE will no longer allow spaceship consoles to be built on non-SE surfaces in order to address this and other crashes.

**Comment by NickMcGettigan** [comment]
There are two of us but we don't think we placed a spaceship console anywhere we shouldn't have, neither of us have unlocked that either.

**Comment by InappropriatePenguin** [comment]
It seems like something got messed up with some zones in your Universe as they appear to be missing hierarchy index values. You can fix the problem by resetting the hierarchy indices using the following command. You'll be able to use the Universe Explorer after that.

```
/c __space-exploration__ Universe.set_hierarchy_values()
```

I'm noticing some of those same zones are _also_ missing `bot_attrition` values, which should have been calculated upon universe creation. This might spam your console with warning messages without crashing.

This too can be addressed, but I'd like to understand how both of these issues came about. I see you have at least three extra (player?) forces in your game—can you detail all the commands you used throughout your game?

**Comment by NickMcGettigan** [comment]
I will load the bad save and try that, thank you.

There were a few of us playing at some point, we knew we wouldn't play together much and wanted a "separate" start.

I believe I used the first command below originally to spawn a few of us but I think I found it was setting all players to the new force, and tried the second when another person joined.  It did seem to cause an issue.  We also never got biter creep.  

Team Creation
/c remote.call("space-exploration", "setup_multiplayer_test", { force_name = "<Player>", players = {game.player}, match_nauvis_seed = true})
/c game.players['<Player>'].force = game.forces['<Player>']
/c game.player.character.die()

Team Creation replace <player> with your ingame name
/c remote.call("space-exploration", "setup_multiplayer_test", { force_name = "<Player>", players = {game.players['<Player>']}, match_nauvis_seed = true})
/c game.players['<Player>'].force = game.forces['<Player>']

We unlocked the research queue after the start and set the evolution to not overwhelm the new players but I believe those are the only commands we used.

**Comment by NickMcGettigan** [comment]
Confirming the command for hierarchy values did allow the universe explorer to open and got a warning about a zone not having a bot attrition value.  The name doesn't look familiar, and it isn't the Nauvis seed which was the starting point for each player.

**Comment by InappropriatePenguin** [comment]
Yeah, this will have to get looked at. Since the vast majority of SE games are either single player or multiplayer-with-a-single-force games, the multiplayer pvp setup command that you were using is not as well tested.

In the mean time, you can fix the missing bot attrition values using the following command:
```lua
/c __space-exploration__ for _, zone in pairs(global.zone_index) do
  if not zone.bot_attrition then
    zone.bot_attrition = Zone.calculate_bot_attrition(zone)
  end
end
```

**Comment by NickMcGettigan** [comment]
We knew it was a bit of a risk but didn't want to schedule a specific day to play so we'd be in sync.  Thanks for your help.
